x/wasm/keeper/keeper.go

```diff
@@ -337,6 +354,11 @@ func (k Keeper) execute(ctx sdk.Context, contractAddress sdk.AccAddress, caller
 	if err != nil {
 		return nil, sdkerrors.Wrap(err, "dispatch")
 	}
+
+	ctx.EventManager().EmitEvent(sdk.NewEvent(
```

Comment:
nice, but same point about ordering. always emit these events *before* the `k.wasmvm.xxx()` calls. All info is available there and this makes sense to prefix the actual wasm messages with the keeper trigger.

that is also how the order is defined in the spec.

Reply:
The order of events was intentionally not addressed as I wanted minimal impact on tests. I should have added this to the todos in the description. sorry.